Pre apprenticeship (#6)
foundersandcoders-members/fac-slides: src/slides/pre-app-1/index.md

---
title: Pre-Apprenticeship Wednesday Week One
---

# Pre-apprenticeship

## Wednesday Week 1

---

<!-- {.primary} -->

Welcome to FAC22.

And congratulations!

---

### Agenda

1. Plan for the next 12 weeks
1. Write a personal user manual
1. Installation
1. Pair programming workshop
1. Daily learning and working outside of this programme
1. For next time

---

## Next 12 weeks

---

### Schedule

- **This week:** Introductions
- **Next week:** No workshops
- **Following weeks**: Two workshops each week

Workshops on Wednesdays and Thursdays from 17:00 - 20:00.
We encourage 10+ hours of self-study outside this programme.

---

### Weekly schedule

- **Wednesday:** Curriculum
- **Thursday:** Employment preparation
- **In your own time:** Study and build websites
- **Throughout:** Maintain a record of your learnings on GitHub

---

## Wednesdays

---

> Wednesdays will focus on bridging the gap between the JavaScript you've learnt while working on your applications and what you'll need to start the course.

Reviewer: Bit more concise, plus doesn't focus exclusively on JS

```suggestion
Bridging the gap between your application learnings and what you'll need on the course.
```

---

> You are responsible for taking charge of your own learning outside the programme. We'll give you a project to work on each week and some resources to help you.

Reviewer: ```suggestion
You're responsible for your own learning outside these sessions. We'll give you a project and some helpful resources each week.
```

You're welcome to pair up with each other outside our workshops. As always, be mindful to be inclusive.

---

## Thursdays

---

> Focused on preparation for interviews with apprenticeship providers. You'll meet employers, follow our employment curriculum, and work together to practice technical interviews.

Reviewer: ```suggestion
Preparing for interviews with apprenticeship providers. You'll meet employers, follow our employment curriculum, and practice technical interviews together.
```

If you're introverted or quiet that's okay too.

We want everyone to have a voice in our discussions and workshops.

---

The onus should not (only) be on quiet people to make themselves heard.

There's limited time and space in groups, especially on Zoom.

Don't use more than your fair share.

---

Please read our Code of Conduct (again).

foundersandcoders.com/code-of-conduct/

Remember this is _the bare minimum_.

It mostly defines unacceptable behaviour, and it's a low bar!

---

> Prioritise marginalised people’s safety over privileged people’s comfort  
> —paraphrased from afrotechfest.co.uk/coc

---

## Zoom etiquette

---

Please rename yourself to the name you'd like to be called and include your pronouns in brackets.

---

If you have a question or would like to volunteer an answer, raise your hand.

---

Be aware of the space you're taking up on Zoom, only one person can speak at once.

---

## User Manuals

---

It's great to have a space to reflect on ourselves, and how we work in a team.

We can also learn about our team-mates before working together, so we know how to get the best out of each other and work well as a team.

---

> Let us know how to say your name and what you'd prefer to be called. If you're happy to share them, include your pronouns too.

Reviewer: We previously said to put pronouns in the Zoom name. If that's optional maybe it should be mentioned there too/instead

---

## Pair programming

---

In pairs, work through the [pair programming workshop](https://learn.foundersandcoders.com/course/handbook/pair-programming/).

---

## Daily learning

---

We believe it's important to build a habit of learning

---

The best way to do this is with structured, focused sessions every (working) day

---

### Active problem-solving

Warm up by writing some code before you start the day

(i.e. CodeWars or Execute Program)

---

### Passive learning

Finish the day by reading an interesting article.

Try to take notes as you go to reinforce the learning.

(Dan likes [Anki](https://apps.ankiweb.net/) for turning notes into flashcards)

---

## Execute Program

---

Execute Program is a great place to practice code

![Execute Program's homepage](../precourse-3/execute-program-1.png)

---

Interactive courses that force you to cement fundamentals

![Execute Program's modern JS course](../precourse-3/execute-program-2.png)

---

Limited number of lessons per day so you can't burn out

---

### Order

1. From today: Modern JavaScript and Arrays
1. From Week 3: Concurrency

> 1. From Week 5: SQL Regular Expressions

Reviewer: ```suggestion
1. From Week 5: SQL and Regular Expressions
```
